src/librustc_data_structures/sorted_map.rs

```diff
@@ -207,8 +218,11 @@ impl<K: Ord, V> SortedMap<K, V> {
 
     /// Looks up the key in `self.data` via `slice::binary_search()`.
     #[inline(always)]
-    fn lookup_index_for(&self, key: &K) -> Result<usize, usize> {
-        self.data.binary_search_by(|&(ref x, _)| x.cmp(key))
+    fn lookup_index_for<Q>(&self, key: &Q) -> Result<usize, usize>
+        where K: Borrow<Q>,
+              Q: Ord + ?Sized
+    {
+        self.data.binary_search_by(|&(ref x, _)| (x.borrow()).cmp(key))
```

Comment:
Micro nit, you can simplify the style to just `self.data.binary_search_by(|(x, _)| x.borrow().cmp(key))`, can't you?

Reply:
True; I'll update shortly.